padertorch/contrib/examples/speaker_classification/supervised/data.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from lazy_dataset.database import JsonDatabase
+
+from padertorch.contrib.je.data.transforms import (
+    LabelEncoder, AudioReader, STFT, MelTransform, Collate
+)
+
+
+def get_datasets(
+    storage_dir, database_json, dataset, batch_size=16, prefetch=True
+):
+    db = JsonDatabase(database_json)
+    ds = db.get_dataset(dataset)
+
+    def prepare_example(example):
+        example['audio_path'] = example['audio_path']['observation']
+        example['speaker_id'] = example['speaker_id'].split('-')[0]
+        return example
+
+    ds = ds.map(prepare_example)
+
+    speaker_encoder = LabelEncoder(
+        label_key='speaker_id', storage_dir=storage_dir, to_array=True
+    )
+    speaker_encoder.initialize_labels(dataset=ds, verbose=True)
+    ds = ds.map(speaker_encoder)
+
+    train_set, validate_set, test_set = train_test_split(ds)
+    training_data = prepare_dataset(train_set, batch_size, training=True)
+    validation_data = prepare_dataset(validate_set, batch_size, training=False)
+    test_data = prepare_dataset(
+        test_set, batch_size, training=False, prefetch=prefetch
+    )
+    return training_data, validation_data, test_data
+
+
+def train_test_split(dataset, dev_split=0.1,  test_split=0.1, seed=0):
+    r = np.random.RandomState(seed)
+
+    try:
+        num_examples = len(dataset)
+    except TypeError:
+        raise RuntimeError('dataset must be indexable!')
+
+    indices = np.arange(num_examples)
+
+    dev_size = int(num_examples * dev_split)
+    test_size = int(num_examples * test_split)
+
+    test_candidates = r.choice(indices, size=test_size, replace=False)
+    indices = np.delete(indices, test_candidates)
+    dev_candidates = r.choice(indices, size=dev_size, replace=False)
+    train_candidates = np.delete(indices, dev_candidates)
+
```

Comment:
why are you creating a test and dev set from train and do not use the original dev and test set?